apis/contour/v1beta1/ingressroute.go

```diff
@@ -141,6 +143,14 @@ type RetryPolicy struct {
 	PerTryTimeout string `json:"perTryTimeout,omitempty"`
 }
 
+// UpstreamValidation defines how to verify the backend service's certificate
+type UpstreamValidation struct {
+	// Name of the Kubernetes secret be used to validate the certificate presented by the backend
+	CACertificate string `json:"caSecret"`
+	// Ket which is expected to be present in the 'subjectAltName' of the presented certificate
+	SubjectName string `json:"subjectName,omitempty"`
```

Comment:
Please remove omit empty, I decided at the last minute that validation of the CA is not sufficient. We'll make both of these parameters mandatory in 0.12